data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -279,31 +291,33 @@ private AccumulatingBulkRequest<BulkOperation, BulkRequest> createBulkRequestFor
         }
     }
 
-    private void handleFailures(final AccumulatingBulkRequest<BulkOperation, BulkRequest> accumulatingBulkRequest, final List<BulkResponseItem> itemResponses) {
+    private void handleFailures(final AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> accumulatingBulkRequest, final List<BulkResponseItem> itemResponses) {
         assert accumulatingBulkRequest.getOperationsCount() == itemResponses.size();
         final ImmutableList.Builder<FailedBulkOperation> failures = ImmutableList.builder();
         for (int i = 0; i < itemResponses.size(); i++) {
             final BulkResponseItem bulkItemResponse = itemResponses.get(i);
-            final BulkOperation bulkOperation = accumulatingBulkRequest.getOperationAt(i);
+            final BulkOperationWrapper bulkOperationWithHandle = (BulkOperationWrapper)accumulatingBulkRequest.getOperationAt(i);
```

Comment:
Is this casting required ? I see the method returns BulkOperationWrapper.

`public BulkOperationWrapper getOperationAt(int index)`